File: Visualizer/Visualizer/Model/XOR.cs
namespace Visualizer.Model
{
    class XOR : BitValue
    {
        private BitValue Input1, Input2;

        public XOR(BitValue in1, BitValue in2)
        {
            Input1 = in1;
            Input2 = in2;
        }

        public override bool Get()
        {
            return Input1.Get() ^ Input2.Get();
        }
    }
}

File: Graphula/Graphula/xor.cpp
#include "xor.h"

XOR::XOR(BitValue * in1, BitValue * in2)
{
    _inputs[0] = in1;
    _inputs[1] = in2;
}


BitValue *XOR::GetInputs()
{
    return _inputs[0];
}

int XOR::GetInputsCount()
{
    return 2;
}

QString XOR::GetName()
{
    return "XOR";
}

bool XOR::Calc()
{
    return _inputs[0]->Get() ^ _inputs[1]->Get();
}
